examples/tdigest/flat.go

```diff
@@ -0,0 +1,39 @@
+package tdigest
+
+import (
+	"sort"
+)
+
+// Flat uses uniform probability density for each centroid.
+type Flat struct {
+	t *TDigest
+}
+
+var _ ProbabilityFunc = &Flat{}
+
+func newFlatFunc(t *TDigest) *Flat {
+	return &Flat{t: t}
+}
+
+func (f *Flat) Probability(value float64) float64 {
+	digest := f.t.Digest
+	sumw := f.t.SumWeight
+	if value <= digest[0].Mean {
+		return digest[0].Weight / (2 * sumw)
+	}
+	if value >= digest[len(digest)-1].Mean {
```

Comment:
The challenge in this code is to estimate the density of buckets outside the range that was covered in the prior window. Here I make the extreme buckets have half the density of their neighbor, which is a bit arbitrary.

The idea is that in order to use inverse-frequency weighted sampling, you need an estimate for what you haven't seen before. For a numerical distribution, the approach here seems to work but isn't perfect.

For a categorical distribution, I've looked into using a non-parametric estimate based on the theory of species-diversity estimation ([see here](http://chao.stat.nthu.edu.tw/wordpress/publication-list/)), which academically derives from [Goode-Turing Frequency Estimation](https://en.wikipedia.org/wiki/Good%E2%80%93Turing_frequency_estimation). This is a curiosity of mine.